packages/fxa-settings/src/pages/InlineRecoverySetup/container.tsx

```diff
@@ -42,61 +38,69 @@ export const InlineRecoverySetupContainer = ({
   );
 
   const location = useLocation() as ReturnType<typeof useLocation> & {
-    state: {
-      totp: TotpToken;
-    };
+    state?: SigninRecoveryLocationState;
   };
-  const totp = location.state?.totp;
+  const signinRecoveryLocationState = location.state;
+  const { totp, ...signinLocationState } = signinRecoveryLocationState || {};
 
   const [recoveryCodes, setRecoveryCodes] = useState<string[]>();
   const [verifyTotp] = useMutation<{ verifyTotp: { success: boolean } }>(
     VERIFY_TOTP_MUTATION
   );
 
+  const { data: totpStatus, loading: totpStatusLoading } =
+    useQuery<TotpStatusResponse>(GET_TOTP_STATUS);
+
   const verifyTotpHandler = useCallback(async () => {
-    const code = await getCode(totp.secret);
+    const code = await getCode(totp!.secret);
     const result = await verifyTotp({
       variables: { input: { code, service: serviceName } },
     });
     return result.data!.verifyTotp.success;
   }, [serviceName, totp, verifyTotp]);
 
   const successfulSetupHandler = useCallback(async () => {
+    // When this is called, we know signinRecoveryLocationState exists.
     const { redirect } = await finishOAuthFlowHandler(
-      account.uid,
-      session.token
+      signinRecoveryLocationState!.uid,
+      signinRecoveryLocationState!.sessionToken,
```

Comment:
I'm wondering if we can remove `sessionToken` from the signin state. Once a user successfully signs in, it's in local storage, and the value in local storage is what we use for GQL requests with `sessionToken()`. I can look at this also in that follow up mentioned.

Reply:
Going to look at this in 9235 since I'm touching some of this there 👍